x/wasm-storage/types/wasm.go

```diff
@@ -0,0 +1,51 @@
+package types
+
+import (
+	fmt "fmt"
+	"strings"
+
+	"github.com/hyperledger/burrow/crypto"
+)
+
+// MaxWasmSize is the maximum size of Wasm bytecode.
+const MaxWasmSize = 800 * 1024
+
+func validateWasmCode(s []byte) error {
+	if len(s) == 0 {
+		return fmt.Errorf("empty Wasm code")
+	}
+	if len(s) > MaxWasmSize {
+		return fmt.Errorf("Wasm code cannot be longer than %d bytes", MaxWasmSize)
+	}
+	return nil
+}
+
+// NewWasm constructs a new Wasm object given bytecode and Wasm type.
+// It panics if it fails to compute hash of bytecode.
+func NewWasm(bytecode []byte, wasmType WasmType) *Wasm {
+	hash := crypto.Keccak256(bytecode)
```

Comment:
Ignore this marker. It's so we know what hash algo to use on the overlay side :).